packages/backend/src/auth/auth.service.ts

```diff
@@ -24,16 +28,25 @@ export class AuthService {
   }
 
   // TODO Fix up these any's soon!
-  async login(user: any) {
-    const payload = { username: user.username, sub: user.userId };
-    return {
-      accessToken: this.jwtService.sign(payload),
-    };
+  async login(user: User | undefined) {
+    if (user) {
+      const payload = { username: user.username };
+      return {
+        accessToken: this.jwtService.sign(payload),
+      };
+    }
+    return null;
   }
 
-  async signup(user: any) {
-    const { username, password } = user;
-    await this.usersService.save({ username, password });
-    return `Signed up username: ${username} with password: ${password}!`;
+  async signup(user: User): Promise<string> {
+    const { username, password, emailAddress } = user;
+    const res = await this.usersService.save({
+      username,
+      usernameLower: username.toLowerCase(),
+      password,
+      emailAddress,
+    });
+    // might want to change this to the response object in the future
```

Comment:
I think if we return a string, Nest JS will automatically put that as the body of the response :). I think this is good to leave in (now that you've taken out the password for me).